packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -115,10 +105,16 @@ library LibProposing {
             {
                 uint256 fee;
                 (newFeeBase, fee, deposit) = getBlockFee(state, config);
-                TaikoToken(resolver.resolve("tko_token", false)).burn(
-                    msg.sender,
-                    fee + deposit
-                );
+
+                uint256 burnAmount = fee + deposit;
+                if (state.balances[msg.sender] > burnAmount) {
+                    state.balances[msg.sender] -= burnAmount;
+                } else {
+                    TaikoToken(resolver.resolve("tko_token", false)).burn(
+                        msg.sender,
+                        burnAmount
+                    );
+                }
```

Comment:
Questions:
- Is the 'fee' is always bigger than the deposit ? So the deposit is just a 'small' amount (currently set 25% of the fee) compared to it. If so, do not usually - in general- the 'deposits' are the larger amounts ? (in terminology)
- In case if proposer has no TKO token at all, how is it punished ? (In current scenario i guess it is ok, since alpha-2 only solo-proposer can propose).

Reply:
1. Deposit == fee * config. proposerDepositPctg. Not sure about what you meant by "the 'deposits' are the larger amounts "
2. Proposer cannot propose a block without sufficient TKO token, but they will never be published.

Reply:
1.: I just meant the 'wording' was a little bit confusing to me. Current config was: 25%. So deposit is 25% of the fee. In my head i usually translates 'deposit' to a large(er) amount while the 'fee' is usually a commission (very low %) this is what confused me a little and wanted to double check.